BootloaderCommonPkg/Library/ConsoleOutLib/ConsoleOutLib.c

```diff
@@ -49,3 +49,57 @@ ConsoleWrite (
 
   return NumberOfBytes;
 }
+
+/**
+  Print Ascii debug message to consoles
+
+  If Format is NULL, then ASSERT().
+
+  @param  Format      The format string for the debug message to print.
+  @param  ...         The variable argument list whose contents are accessed
+                      based on the format string specified by Format.
+
+**/
+UINTN
+EFIAPI
+ConsoleDebugPrint (
+  IN  CONST CHAR8          *Format,
+  ...
+  )
+{
+  CHAR8    Buffer[MAX_MESSAGE_LENGTH];
+  VA_LIST  Marker;
+
+  VA_START (Marker, Format);
+  AsciiVSPrint (Buffer, sizeof (Buffer), Format, Marker);
+  VA_END (Marker);
+
+  return ConsoleWrite ((UINT8 *)Buffer, AsciiStrLen (Buffer));
+}
+
+/**
+  Print Unicode debug message to consoles
+
+  If Format is NULL, then ASSERT().
+
+  @param  Format      The format string for the debug message to print.
+  @param  ...         The variable argument list whose contents are accessed
+                      based on the format string specified by Format.
+
+**/
+UINTN
+EFIAPI
+ConsoleDebugPrintUnicode (
+  IN  CONST CHAR16         *Format,
+  ...
+  )
+{
+  CHAR8    Buffer[MAX_MESSAGE_LENGTH];
+  VA_LIST  Marker;
+
+  VA_START (Marker, Format);
+  AsciiVSPrintUnicodeFormat (Buffer, sizeof (Buffer), Format, Marker);
+  VA_END (Marker);
+
+  return ConsoleWrite ((UINT8 *)Buffer, AsciiStrLen (Buffer));

```

Comment:
You can use  AsciiVSPrintUnicodeFormat return values for the length of the Buffer.  It can avoid calling AsciiStrLen().

Reply:
Let me use the return length. Thanks.